sql/expression/comparison.go

```diff
@@ -205,6 +209,59 @@ func (e *Equals) DebugString() string {
 	return fmt.Sprintf("%s = %s", sql.DebugString(e.Left()), sql.DebugString(e.Right()))
 }
 
+// NullSafeEquals is a comparison that checks an expression is equal to
+// another, where NULLs do no coalesce to NULL and two NULLs compare equal to
+// each other.
+type NullSafeEquals struct {
+	comparison
+}
+
+// NewNullSafeEquals returns a new Equals expression.
+func NewNullSafeEquals(left sql.Expression, right sql.Expression) *NullSafeEquals {
+	return &NullSafeEquals{newComparison(left, right)}
+}
+
+// Type implements the Expression interface.
+func (e *NullSafeEquals) Type() sql.Type {
+	return sql.Int8
+}
+
+// Eval implements the Expression interface.
+func (e *NullSafeEquals) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
+	result, err := e.Compare(ctx, row)
+	if err != nil {
+		if ErrNilOperand.Is(err) {
```

Comment:
This is gross, you should just override the Compare method for NullSafeEqual and then treat all errors as unrecoverable